packages/@aws-cdk/aws-pipes-sources-alpha/lib/dynamodb.ts

```diff
@@ -0,0 +1,158 @@
+import { IPipe, ISource, SourceConfig } from '@aws-cdk/aws-pipes-alpha';
+import { Duration } from 'aws-cdk-lib';
+import { ITableV2 } from 'aws-cdk-lib/aws-dynamodb';
+import { IRole } from 'aws-cdk-lib/aws-iam';
+import { DeadLetterConfigParameters } from './deadLetterConfig';
+import { DynamoDBStartingPosition, OnPartialBatchItemFailure } from './enums';
+
+/**
+ * Parameters for the DynamoDB source.
+ */
+export interface DynamoDBSourceParameters {
+  /**
+    * The maximum number of records to include in each batch.
+    *
+    * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipesourcedynamodbstreamparameters.html#cfn-pipes-pipe-pipesourcedynamodbstreamparameters-batchsize
+    * @default 1
+    */
+  readonly batchSize?: number;
+
+  /**
+    * Define the target queue to send dead-letter queue events to.
+    *
+    * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipesourcedynamodbstreamparameters.html#cfn-pipes-pipe-pipesourcedynamodbstreamparameters-deadletterconfig
+    * @default no dead letter queue
+    */
+  readonly deadLetterConfig?: DeadLetterConfigParameters;
+
+  /**
+   * The maximum length of a time to wait for events.
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipesourcedynamodbstreamparameters.html#cfn-pipes-pipe-pipesourcedynamodbstreamparameters-maximumbatchingwindowinseconds
+   * @default no batching window
+   */
+  readonly maximumBatchingWindow?: Duration;
+
+  /**
+   * (Streams only) Discard records older than the specified age. The default value is -1, which sets the maximum age to infinite. When the value is set to infinite, EventBridge never discards old records.
+   *
+   * Leave undefined to set the maximum record age to infinite.
+   *
+   *  @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipesourcedynamodbstreamparameters.html#cfn-pipes-pipe-pipesourcedynamodbstreamparameters-maximumrecordageinseconds
+   * @default -1 (infinite)
+   */
+  readonly maximumRecordAge?: Duration;
```

Comment:
`@default undefined` here as well, see below

Reply:
I had `undefined` originally but the linter didn't like it. It wants the value CFN will assign, even if we don't assign -1 below.